flight/mcu/src/mcu_ac/main.cpp

```diff
@@ -146,18 +179,19 @@ static THD_FUNCTION(highgIMU_THD, arg) {
     // Load outside variables into the function
     struct pointers *pointer_struct = (struct pointers *)arg;
 
-    KX134 *highG = pointer_struct->highGimuPointer;
+    QwiicKX134 *highG = pointer_struct->highGimuPointer;
     DataLogBuffer *data_log_buffer = &pointer_struct->dataloggerTHDVarsPointer;
     HighGData *highg_data = &pointer_struct->sensorDataPointer->highG_data;
 
     while (true) {
 #ifdef THREAD_DEBUG
         Serial.println("### High G IMU thread entrance");
 #endif
-
+        chMtxLock(&data_log_buffer->dataMutex_highG);
         highGimuTickFunction(highG, data_log_buffer, highg_data);
+        chMtxUnlock(&data_log_buffer->dataMutex_highG);
 
-        chThdSleepMilliseconds(6);
+        chThdSleepMilliseconds(20);  // highg reads at 50 hz
```

Comment:
Any particular reason why we run the KX134 sensor at 50hz? I've gotten it to work at a 10Khz data output rate

Reply:
@NPIPHI not sure about this one. Thoughts?

Reply:
physical sensor limitation

Reply:
That's what I thought, but rishi said he's got the kx to run at a 10khz output rate

Reply:
Oh wait nvm he edited his comment. Still, [this page](https://www.sparkfun.com/products/17589#:~:text=The%20KX134%20is%20a%20low,applications%20such%20as%20vibration%20sensing.) confirms that?

Reply:
its configurable up to that much, 50hz is the default setting. We could raise the input rate at startup though